File: python/unidad_02/problema_2_11.py
'''
El programa, dados como dato la clave de la zona a la cual se llamó y el
número de minutos que duró la llamada, calcula el costo total de la misma.
'''

def calcular_costo(clave, minutos):

    costo = 0

    if clave == 12:

        costo = minutos * 2

    elif clave == 15:

        costo = minutos * 2.2

    elif clave == 18:

        costo = minutos * 4.5

    elif clave == 19:

        costo = minutos * 3.5

    elif clave == 23 or clave == 25:

        costo = minutos * 6

    elif clave == 29:

        costo = minutos * 5

    else:

        print("Error: clave no válida.")

    return costo

def main():

    try:

        clave = int(input("Clave telefónica: "))

        minutos = int(input("Duración de la llamada (min): "))

        if minutos <= 0:

            print("Error: la duración debe ser mayor que cero.")
            return

    except ValueError as e:

        print(e)
        return
    
    costo = calcular_costo(clave, minutos)

    if costo != 0:
    
        print(f"El costo de llamada es de {costo} u")

if __name__ == '__main__':

    main()

File: java/unidad_02/Problema_2_11/src/Problema_2_11.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * El programa, dados como dato la clave de la zona a la cual se llamó y el
 * número de minutos que duró la llamada, calcula el costo total de la misma.
 */

public class Problema_2_11 {

    public static void calcularCosto(int clave, int minutos) {

        float costo;

        switch (clave) {

            case 12:

                costo = minutos * 2;
                break;

            case 15:

                costo = minutos * (float) 2.2;
                break;

            case 18:

                costo = minutos * (float) 4.5;
                break;

            case 19:

                costo = minutos * (float) 3.5;
                break;

            case 23:
            case 25:

                costo = minutos * 6;
                break;

            case 29:

                costo = minutos * 5;
                break;

            default:

                System.out.println("Error: clave no válida.");
                return;
        }
        System.out.printf("El costo de llamada es de %.2f u", costo);
    }

    public static void main(String[] args) {

        int clave, minutos;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("Clave telefónica: ");
            clave = scanner.nextInt();

            System.out.print("Duración de llamada (min): ");
            minutos = scanner.nextInt();

            if (minutos <= 0) {

                System.out.println("Error: la duración debe ser mayor que cero.");
                return;
            }
        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        calcularCosto(clave, minutos);
    }
}